controllers/provisioners/eks/helpers.go

```diff
@@ -890,7 +891,7 @@ func (ctx *EksInstanceGroupContext) GetOverrides() []*autoscaling.LaunchTemplate
 	)
 	overrides := []*autoscaling.LaunchTemplateOverrides{}
 
-	if mixedPolicy.InstanceTypes != nil {
+	if mixedPolicy != nil && mixedPolicy.InstanceTypes != nil {
```

Comment:
Mentioned this in slack; I hit a case pretty early on where `mixedPolicy` was `nil` - maybe resolved with the latest changes.

Reply:
mixedPolicy is guaranteed to be non nil inside GetOverrides due to:
https://github.com/keikoproj/instance-manager/blob/master/controllers/provisioners/eks/helpers.go#L930-L934

Reply:
not sure I'm following. `mixedPolicy` comes from `configuration.GetMixedInstancesPolicy()`

https://github.com/keikoproj/instance-manager/blob/6f9289fecaf6793a42a399cf902e1f18670082d0/api/v1alpha1/instancegroup_types.go#L593-L595

which can be nil?

Reply:
Correct, but `GetDesiredMixedInstancesPolicy` calls `GetOverrides` and does a nil check.
Eitherway, this can probably stay / won't hurt.